python/ray/data/read_api.py

```diff
@@ -698,6 +704,113 @@ def read_parquet(
     )
 
 
+@PublicAPI
+def read_delta(
+    uri: str,
+    delta_package: str = None,
+    *,
+    filesystem: Optional["pyarrow.fs.FileSystem"] = None,
+    columns: Optional[List[str]] = None,
+    parallelism: int = -1,
+    ray_remote_args: Dict[str, Any] = None,
+    tensor_column_schema: Optional[Dict[str, Tuple[np.dtype, Tuple[int, ...]]]] = None,
+    meta_provider: Optional[ParquetMetadataProvider] = None,
+    **arrow_parquet_args,
+) -> Dataset:
+    """Creates a :class:`~ray.data.Dataset` from delta files.
+
+
+    Examples:
+        Read delta files.
+
+        >>> import ray
+        >>> ds = ray.data.read_delta("/path/to/your/delta/file")
+
+        The Delta reader requires a spark session to be created.
+
+        .. testcode::
+
+            import ray
+            from pyspark.sql import SparkSession
+
+            spark = (SparkSession.builder
+            .config("spark.jars.packages", "io.delta:delta-core_2.12:2.2.0")
+            .config("spark.sql.extensions", "io.delta.sql.DeltaSparkSessionExtension")
+            .config("spark.sql.catalog.spark_catalog",
+                    "org.apache.spark.sql.delta.catalog.DeltaCatalog")
+            .getOrCreate())
+
+            # Create a delta table using spark
+            spark_df = spark.range(200).repartition(8)
+            spark_df.write.format("delta").save("DeltaTable")
+
+            # Use ray to read the delta table
+            data = ray.data.read_delta("DeltaTable")
+            assert data.count() == 200
+            assert data.columns() == ["id"]
+            assert data.schema().names == ["id"]
+            assert data.schema().types[0] == "int64"
+
+
+    Args:
+        uri: A single file path or directory.
```

Comment:
You should document on databricks, only dbfs path is supported, and `filesystem` argument is igored.

Reply:
I rethink about it,
on databricks, besides supporting dbfs, we can still support s3/hdfs/.. etc similar OSS case.

Reply:
So should we check if it's on databricks, we set this variable to None?

Reply:
We can update `_convert_dbfs_path_to_local_path` to only convert dbfs path, for other scheme, return unchanged path